src/fs/config.rs

```diff
@@ -0,0 +1,43 @@
+use crate::sys::*;
+
+pub struct Configuration {
+    pub max_files: u32,
+    pub format_if_mount_failed: bool,
+    pub allocation_unit_size: usize,
+    #[cfg(not(esp_idf_version_major = "4",))] // For ESP-IDF v5.0 and later
+    pub disk_status_check_enable: bool,
+}
+
+impl Default for Configuration {
+    fn default() -> Self {
+        Self {
+            max_files: 4,
+            format_if_mount_failed: false,
+            allocation_unit_size: 16 * 1024,
+            #[cfg(not(
+                esp_idf_version_major = "4",
+            ))] // For ESP-IDF v5.0 and later
+            disk_status_check_enable: false,
+        }
+    }
+}
+
+impl Configuration {
+    pub fn new() -> Self {
```

Comment:
Make this `const fn` and move the implementation _here_. Then have `Default::default` delegate to the `const fn` constructor. This way the user can construct configuration in a co st context too, if she wants to.